app/models/competition.rb

```diff
@@ -2490,23 +2499,33 @@ def set_form_data(form_data, current_user)
         if %w[registration.closingDateTime registration.waitingListDeadlineDate registration.eventChangeDeadlineDate].include?(joined_key)
           existing_value = current_state_form.dig(*prefixes, key)
 
-          if existing_value.present?
-            existing_datetime = DateTime.parse(existing_value).utc
+          previously_had_value = existing_value.present?
+          will_have_value = value.present?
 
-            raise WcaExceptions::BadApiParameter.new(I18n.t('competitions.errors.editing_deadline_already_passed', timestamp: existing_datetime), json_property: joined_key) unless existing_datetime >= DateTime.now.utc
+          # Complain if the existing timestamp lies in the past
+          #   Note: Some timestamps are less strict than others (see https://github.com/thewca/worldcubeassociation.org/issues/11416)
+          #   so we only enforce this validation on a subset of timestamps.
+          if previously_had_value && %w[registration.closingDateTime registration.waitingListDeadlineDate].include?(joined_key)
```

Comment:
Why not split this list out to a variable/constant declaration? That would mean we can explicitly call it "can be edited if past" or whatever, meaning we don't need to infer the significance of this list